File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/c_solutions/2734-lexicographically-smallest-string-after-substring-operation.c
char* smallestString(char* s) {
    int n = strlen(s);
    int i = 0;

    while (i < n && s[i] == 'a') {
        i++;
    }

    if (i == n) {
        s[n - 1] = 'z';
        return s;
    }

    while (i < n && s[i] != 'a') {
        s[i] = s[i] - 1;
        i++;
    }

    return s;
}

File: leetcode/AIcoding/3.DeepSeek-v3/rust_solutions/2734-lexicographically-smallest-string-after-substring-operation.rs
impl Solution {
    pub fn smallest_string(s: String) -> String {
        let mut s_bytes = s.into_bytes();
        let mut i = 0;
        while i < s_bytes.len() && s_bytes[i] == b'a' {
            i += 1;
        }
        if i == s_bytes.len() {
            s_bytes[s_bytes.len() - 1] = b'z';
            return String::from_utf8(s_bytes).unwrap();
        }
        let mut j = i;
        while j < s_bytes.len() && s_bytes[j] != b'a' {
            s_bytes[j] -= 1;
            j += 1;
        }
        String::from_utf8(s_bytes).unwrap()
    }
}